evaluation.md

```diff
@@ -5,15 +5,42 @@
 _Describe each change from your ideal syntax to the syntax you implemented, and
 describe_ why _you made the change._
 
+
+The first change from our ideal syntax to the syntax we implemented was to also make the end state of a rule a method that takes in an argument to be consistent with the other aspects of a rule. Because our DSL targetted those unfamiliar with computer science, it made sense to make the method-argument model consistent throughout. Another early change from our ideal syntax (although this is more on implementation details than syntactical appearance) was using ShortRule, our own defined class, rather than Rule. We reasoned that because we were abstracting out the start state from the rule, we could just use ShortRule, which only took in surroundings, moveDirection, and endState, and then construct a Rule from our state method by passing in the startState. We also realized that it was probably impossible to not pass the state name to a state method, so we made our state method take in a string that would be implicitly converted to a State type and a variatic sequences of Rules. The move and to methods were very straightforward to implement: the move method took in a moveDirection and returned that moveDirection and the to method took in an string (which was implicitly converted to a State), and returned the endState. The open and blocked open took in a variatic variable of moveDirections and used pattern-matching to determine the corresponding surroundings. The most difficult part was implementing the binary operator to combine open and blocked so that only one Surrounding would be returned. Here's a snippet of the comments we had for implementing the binary operator (to give some insight on how we approached it):
+
+Notes on combining blocked + open (from Tues night):
+ * can combine with "+" in the form "this.+(open(..), blocked(..))" without making our own wrapper class (simply unpack Surroundings and create new one),
+ * However, to get proper binary operator (with the form, "open(..) + blocked(..)", we need to create our own class that mimics Surrounding because we'd need to define the operator inside it.
+ * (^Unless I'm forgetting about something obvious)
+ * Now our tuples might contain Surroundings or OurSurroundings (if users only pass in an 'open', for example, the '+' method will never be called, so the OurSurroundings will never get converted to a Surroundings).
+ * So, we must cast both types of tuples to ShortRule.
+
+As the comments above show, we needed to create a wrapper class OurSurroundings to get the binary operator working, which was something we also didn't expect when writing our ideal example.
```

Comment:
Was OurSurroundings useful for any other part of your organization, or was it just necessary for this one problem?